Removes Mautic configuration's dependency on the container

alanhartless opened the pull request:
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | Y
| Automated tests included? | Yes and more Coming
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/8316
| BC breaks? | 
| Deprecations? | Yes

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR removes Mautic's dependency on Symfomy's container for its configuration options using Symfony 3's support for environment variables. This means that the Container no longer needs to be rebuilt when Mautic's configuration UI is updated. There are a few exceptions that require editing of the local config file such as API rate limiting that still requires rebuilding the cache. But the goal is that anything that can be edited through the configuration UI should not require a rebuild of the container.  

Unfortunately this is big and touches several major areas due to the need to refactor away from cache compiler passes. 

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Mautic should load :-D
3. Test that the Configuration saves and the values are persisted to app/config/local.php
4. Test sending emails both immediately and spooled to the filesystem
5. Test a SAML implementation (https://www.mautic.org/docs/en/authentication/saml.html; I personally tested using SF as an identify provider https://help.salesforce.com/articleView?id=identity_provider_enable.htm&type=5). If using Salesforce, configure the SAML Email attribute to be "Username", First name = "FirstName" and Last name = "LastName")
6. Test the QueueBundle (need help on the steps for this). The UI can no longer be used for this. You have to edit app/config/local.php manually. See https://github.com/mautic/mautic/pull/8265/commits/f6941e808bb4abd348034aad47dffb858c81d152
7. Test Mautic's installer
8. Test that API endpoints (including auth endpoints) are blocked if API is disabled. Also test that basic auth is rejected if disabled.
9. Test installing a new language through Configuration; log out and back in; the new language should be loaded
10. Test reloading plugins; clear out the plugins table then run `php bin/console -e prod mautic:plugins:reload`. Check /s/plugins that they are again listed.
11. Add `'debug' => true,` to app/config/local.php. Run the command to update the segment `php bin/console mautic:segments:update -e prod`. Check var/logs to see if there is a mautic_prod-YYYY-MM-DD.php file with debug info for building the segments. Now change debug to false and run the command again and no new DEBUG output should be written to the log. 
12. Test El Finder https://github.com/mautic/mautic/pull/8306
13. Send an email that contains links to contacts via CLI (campaigns are currently broken in 3.x so I had to use the `mautic:broadcast:send` command with a broadcast email that had a publish at and down date configured). The URLs should be correct based on what's configured as the `site_url` in app/config/local.php 

#### List backwards compatibility breaks:
1. See UPGRADE-3.0.md

hluchas commented:
Prod cache is not warming up 
[warmup.log](https://github.com/mautic/mautic/files/4117153/warmup.log)

alanhartless commented:
@hluchas those two issues have been resolved. Thanks!

hluchas commented on app/config/config_prod.php:
```diff
@@ -24,18 +24,18 @@
             'type'         => 'fingers_crossed',
             'buffer_size'  => '200',
-            'action_level' => ($debugMode) ? 'debug' : 'error',
+            'action_level' => $debugMode ? 'debug' : '%env(MAUTIC_LOG_MAIN_ACTION_LEVEL)%',
```
This is producing 
```
~/dev/community-fork(containerless-config) $ bin/console -v

In Compiler.php line 157:

  [Symfony\Component\DependencyInjection\Exception\EnvParameterException]
  Incompatible use of dynamic environment variables "MAUTIC_LOG_MAIN_ACTION_LEVEL" found in parameters.


Exception trace:
  at /Users/lukas.drahy/dev/community-fork/vendor/symfony/dependency-injection/Compiler/Compiler.php:157
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/dependency-injection/ContainerBuilder.php:789
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/http-kernel/Kernel.php:643
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /Users/lukas.drahy/dev/community-fork/app/AppKernel.php:276
 AppKernel->boot() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/framework-bundle/Console/Application.php:64
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php:148
 Symfony\Component\Console\Application->run() at /Users/lukas.drahy/dev/community-fork/bin/console:41

In MonologExtension.php line 53:

  [InvalidArgumentException]
  Undefined constant `Monolog\Logger::%env(MAUTIC_LOG_MAIN_ACTION_LEVEL)%` Argument `%env(MAUTIC_LOG_MAIN_ACTION_LEVEL)%` passed


Exception trace:
  at /Users/lukas.drahy/dev/community-fork/vendor/symfony/monolog-bundle/DependencyInjection/MonologExtension.php:53
 Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension->levelToMonologConst() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/monolog-bundle/DependencyInjection/MonologExtension.php:389
 Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension->buildHandler() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/monolog-bundle/DependencyInjection/MonologExtension.php:84
 Symfony\Bundle\MonologBundle\DependencyInjection\MonologExtension->load() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/dependency-injection/Compiler/MergeExtensionConfigurationPass.php:71
 Symfony\Component\DependencyInjection\Compiler\MergeExtensionConfigurationPass->process() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/http-kernel/DependencyInjection/MergeExtensionConfigurationPass.php:39
 Symfony\Component\HttpKernel\DependencyInjection\MergeExtensionConfigurationPass->process() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/dependency-injection/Compiler/Compiler.php:140
 Symfony\Component\DependencyInjection\Compiler\Compiler->compile() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/dependency-injection/ContainerBuilder.php:789
 Symfony\Component\DependencyInjection\ContainerBuilder->compile() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/http-kernel/Kernel.php:643
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /Users/lukas.drahy/dev/community-fork/app/AppKernel.php:276
 AppKernel->boot() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/framework-bundle/Console/Application.php:64
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php:148
 Symfony\Component\Console\Application->run() at /Users/lukas.drahy/dev/community-fork/bin/console:41

~/dev/community-fork(containerless-config) $
```

hluchas commented:
Testing:
- [x] Test that the Configuration saves and the values are persisted to app/config/local.php
- [x] CLI commands for segments, campaigns, plugins
- [x] Test sending emails both immediately and spooled to the filesystem
- [] Test a SAML implementation
- [] Test the QueueBundle
- [] Test Mautic's installer
- [] Test that API endpoints

fixes here : https://github.com/alanhartless/mautic/pull/9

alanhartless commented:
@hluchas I fixed the logging issue and installer issues. I also found an issue with languages still getting compiled into the container and some places that tried to get a plugin list from CoreParametersHelper which is no longer possible. I also found an issue that container parameters in values of Mautic config values were not getting resolved so fixed that. 

I added a couple extra testing steps to the PRs description. 

For the `Router broken` you have, that seems to be more related to your local setup. Is your .htaccess getting picked up?

hluchas commented:
How to test QueueBunlde:
in local.php : 
```
'queue_mode' => 'command_process',
'queue_protocol' => 'beanstalkd',
```
bin/console cache:warmup -e dev
bin/console mautic:queue:process --queue-name mautic -e dev -v
in local.php : 
```
'queue_mode' => 'command_process',
'queue_protocol' => 'beanstalkd',
```
bin/console cache:warmup -e dev
bin/console mautic:queue:process --queue-name mautic -e dev -v

Processing webhooks and tracking:
TODO

hluchas commented:
```/usr/local/Cellar/php@7.2/7.2.27/bin/php /Users/lukas.drahy/dev/community-fork/bin/console mautic:queue:process --queue-name lead -e dev -v
12:09:46 NOTICE    [mautic] Pheanstalk\Exception\SocketException: Socket timed out! (uncaught exception) at /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Socket/NativeSocket.php line 115 while running console command `mautic:queue:process`
[stack trace]
#0 /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Connection.php(111): Pheanstalk\Socket\NativeSocket->getLine()
#1 /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Pheanstalk.php(398): Pheanstalk\Connection->dispatchCommand(Object(Pheanstalk\Command\ReserveCommand))
#2 /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Pheanstalk.php(280): Pheanstalk\Pheanstalk->_dispatch(Object(Pheanstalk\Command\ReserveCommand))
#3 /Users/lukas.drahy/dev/community-fork/vendor/leezy/pheanstalk-bundle/src/Proxy/PheanstalkProxy.php(293): Pheanstalk\Pheanstalk->reserve(3600)
#4 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/EventListener/BeanstalkdSubscriber.php(74): Leezy\PheanstalkBundle\Proxy\PheanstalkProxy->reserve(3600)
#5 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/EventListener/AbstractQueueSubscriber.php(53): Mautic\QueueBundle\EventListener\BeanstalkdSubscriber->consumeMessage(Object(Mautic\QueueBundle\Event\QueueEvent))
#6 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(115): Mautic\QueueBundle\EventListener\AbstractQueueSubscriber->onConsumeMessage(Object(Mautic\QueueBundle\Event\QueueEvent), 'mautic.queue_co...', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
#7 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/EventDispatcher.php(214): Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(Object(Mautic\QueueBundle\Event\QueueEvent), 'mautic.queue_co...', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
#8 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'mautic.queue_co...', Object(Mautic\QueueBundle\Event\QueueEvent))
#9 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(143): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('mautic.queue_co...', Object(Mautic\QueueBundle\Event\QueueEvent))
#10 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/Queue/QueueService.php(80): Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch('mautic.queue_co...', Object(Mautic\QueueBundle\Event\QueueEvent))
#11 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/Command/ConsumeQueueCommand.php(79): Mautic\QueueBundle\Queue\QueueService->consumeFromQueue('lead', NULL)
#12 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Command/Command.php(255): Mautic\QueueBundle\Command\ConsumeQueueCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php(1001): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /Users/lukas.drahy/dev/community-fork/vendor/symfony/framework-bundle/Console/Application.php(86): Symfony\Component\Console\Application->doRunCommand(Object(Mautic\QueueBundle\Command\ConsumeQueueCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php(255): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(Mautic\QueueBundle\Command\ConsumeQueueCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /Users/lukas.drahy/dev/community-fork/vendor/symfony/framework-bundle/Console/Application.php(74): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php(148): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 /Users/lukas.drahy/dev/community-fork/bin/console(41): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 {main}
12:09:46 ERROR     [console] Error thrown while running command "mautic:queue:process --queue-name lead -e dev -v". Message: "Socket timed out!" ["exception" => Pheanstalk\Exception\SocketException { …},"command" => "mautic:queue:process --queue-name lead -e dev -v","message" => "Socket timed out!"]
12:09:46 WARNING   [mautic] Command `mautic:queue:process` exited with status code 1
```

escopecz commented:
Conflicts resolved by merging the latest 3.x branch into this one.

hluchas commented:
GET http://mautic-fork.local/api/assets
```
127.0.0.1 - - [03/Feb/2020:14:03:31 +0100] "GET /api/assets HTTP/1.1" 302 229
127.0.0.1 - admin [03/Feb/2020:14:03:31 +0100] "GET /index.php/api/assets HTTP/1.1" 301 438
```
No response received

Doesn't work with `index_dev.php`  too

```
127.0.0.1 - - [03/Feb/2020:14:09:18 +0100] "GET /index_dev/api/assets HTTP/1.1" 302 239
127.0.0.1 - admin [03/Feb/2020:14:09:18 +0100] "GET /index.php/index_dev/api/assets HTTP/1.1" 301 478
```

escopecz commented:
I can't load Mautic on this branch due to `InvalidArgumentException: Not a built-in Swiftmailer transport: mautic.transport.sendgrid. in /vendor/symfony/swiftmailer-bundle/DependencyInjection/SwiftmailerTransportFactory.php:74` error. I can see that such transport should exist.

alanhartless commented:
@escopecz https://github.com/mautic/mautic/pull/8265/commits/6f521ab014b14ff7c485232966a5eee7da06ed37 should fix https://github.com/mautic/mautic/pull/8265#issuecomment-581416960

alanhartless commented:
> ```
> /usr/local/Cellar/php@7.2/7.2.27/bin/php /Users/lukas.drahy/dev/community-fork/bin/console mautic:queue:process --queue-name lead -e dev -v
> 12:09:46 NOTICE    [mautic] Pheanstalk\Exception\SocketException: Socket timed out! (uncaught exception) at /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Socket/NativeSocket.php line 115 while running console command `mautic:queue:process`
> [stack trace]
> #0 /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Connection.php(111): Pheanstalk\Socket\NativeSocket->getLine()
> #1 /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Pheanstalk.php(398): Pheanstalk\Connection->dispatchCommand(Object(Pheanstalk\Command\ReserveCommand))
> #2 /Users/lukas.drahy/dev/community-fork/vendor/pda/pheanstalk/src/Pheanstalk.php(280): Pheanstalk\Pheanstalk->_dispatch(Object(Pheanstalk\Command\ReserveCommand))
> #3 /Users/lukas.drahy/dev/community-fork/vendor/leezy/pheanstalk-bundle/src/Proxy/PheanstalkProxy.php(293): Pheanstalk\Pheanstalk->reserve(3600)
> #4 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/EventListener/BeanstalkdSubscriber.php(74): Leezy\PheanstalkBundle\Proxy\PheanstalkProxy->reserve(3600)
> #5 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/EventListener/AbstractQueueSubscriber.php(53): Mautic\QueueBundle\EventListener\BeanstalkdSubscriber->consumeMessage(Object(Mautic\QueueBundle\Event\QueueEvent))
> #6 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/Debug/WrappedListener.php(115): Mautic\QueueBundle\EventListener\AbstractQueueSubscriber->onConsumeMessage(Object(Mautic\QueueBundle\Event\QueueEvent), 'mautic.queue_co...', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
> #7 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/EventDispatcher.php(214): Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(Object(Mautic\QueueBundle\Event\QueueEvent), 'mautic.queue_co...', Object(Symfony\Component\HttpKernel\Debug\TraceableEventDispatcher))
> #8 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/EventDispatcher.php(44): Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(Array, 'mautic.queue_co...', Object(Mautic\QueueBundle\Event\QueueEvent))
> #9 /Users/lukas.drahy/dev/community-fork/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php(143): Symfony\Component\EventDispatcher\EventDispatcher->dispatch('mautic.queue_co...', Object(Mautic\QueueBundle\Event\QueueEvent))
> #10 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/Queue/QueueService.php(80): Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch('mautic.queue_co...', Object(Mautic\QueueBundle\Event\QueueEvent))
> #11 /Users/lukas.drahy/dev/community-fork/app/bundles/QueueBundle/Command/ConsumeQueueCommand.php(79): Mautic\QueueBundle\Queue\QueueService->consumeFromQueue('lead', NULL)
> #12 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Command/Command.php(255): Mautic\QueueBundle\Command\ConsumeQueueCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #13 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php(1001): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #14 /Users/lukas.drahy/dev/community-fork/vendor/symfony/framework-bundle/Console/Application.php(86): Symfony\Component\Console\Application->doRunCommand(Object(Mautic\QueueBundle\Command\ConsumeQueueCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #15 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php(255): Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand(Object(Mautic\QueueBundle\Command\ConsumeQueueCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #16 /Users/lukas.drahy/dev/community-fork/vendor/symfony/framework-bundle/Console/Application.php(74): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #17 /Users/lukas.drahy/dev/community-fork/vendor/symfony/console/Application.php(148): Symfony\Bundle\FrameworkBundle\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #18 /Users/lukas.drahy/dev/community-fork/bin/console(41): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #19 {main}
> 12:09:46 ERROR     [console] Error thrown while running command "mautic:queue:process --queue-name lead -e dev -v". Message: "Socket timed out!" ["exception" => Pheanstalk\Exception\SocketException { …},"command" => "mautic:queue:process --queue-name lead -e dev -v","message" => "Socket timed out!"]
> 12:09:46 WARNING   [mautic] Command `mautic:queue:process` exited with status code 1
> ```

@hluchas What is in your local.php config for your queue settings? 
```
// This is an advanced setup allowing a work queue/message broker to process page hits and email tokens outside of the web request.
        // The work queue/message broker must be configured and running outside of Mautic for this to function.
        // Currently supports rabbitmq or beanstalkd
        'queue_protocol'     => '',
        // The hostname of the RabbitMQ server
        'rabbitmq_host'      => 'localhost',
        // The port that the RabbitMQ server is listening on
        'rabbitmq_port'      => '5672',
        // The virtual host to use for this RabbitMQ server
        'rabbitmq_vhost'     => '/',
        // The username for the RabbitMQ server
        'rabbitmq_user'      => 'guest',
        // The password for the RabbitMQ server
        'rabbitmq_password'  => 'guest',
        // The hostname of the Beanstalkd server
        'beanstalkd_host'    => 'localhost',
        // The port that the Beanstalkd server is listening on
        'beanstalkd_port'    => '11300',
        // The default TTR for Beanstalkd jobs
        'beanstalkd_timeout' => '60',
```

Do you have beanstalkd running? Does your local.php's values match beanstalkd's host and port? 

I'm able to connect fine when I configure Mautic to use beanstalkd and run the same command as you. It starts to listen for beanstalkd jobs without throwing any exceptions.

alanhartless commented:
@hluchas https://github.com/mautic/mautic/pull/8265/commits/3e92fd91f066f1486a30e7f31993f6980a094e51 should resolve https://github.com/mautic/mautic/pull/8265#issuecomment-581405939

hluchas commented:
> @hluchas [3e92fd9](https://github.com/mautic/mautic/commit/3e92fd91f066f1486a30e7f31993f6980a094e51) should resolve [#8265 (comment)](https://github.com/mautic/mautic/pull/8265#issuecomment-581405939)

That's it. TY!
I started with the same fix, but the chain of listeners was difficult to debug for me.

hluchas commented:
@alanhartless ad Beanstalk 
my config:
```
    'queue_mode' => 'command_process',
    'queue_protocol' => 'beanstalkd',
```
`bin/console mautic:queue:process --queue-name lead -e dev -v`
Is producing after some time 
`Pheanstalk\Exception\SocketException: Socket timed out! at pda/pheanstalk/src/Socket/NativeSocket.php line 115`

problem is described here  https://github.com/pheanstalk/pheanstalk/issues/180

escopecz commented:
I tested configuration save. It works well. The only difference when I compare the old and the new local.php is the code style change. `array()` became `[]`, arrows are not in one column, commas were removed from last elements of an array. These 4 params were added:
```
        'max_log_files' => 7,
	'log_file_name' => 'mautic_dev.php',
	'track_private_ip_ranges' => false,
	'debug' => false,
```
I also tested installation and it went well. Even with queued email sending.

And sending emails from custom email transports works. I tested Sparkpost. Once Lukas solves his issues, I think this is good to go 👍
